classes/class-wc-connect-taxjar-integration.php

```diff
@@ -104,9 +111,93 @@ public function add_tax_settings( $tax_settings ) {
 		// Insert the "automated taxes" setting at the top (under the section title)
 		array_splice( $tax_settings, 1, 0, array( $automated_taxes ) );
 
+		if ( $this->is_enabled() ) {
+			// If the automated taxes are enabled, disable the settings that would be reverted in the original plugin
+			foreach ( $tax_settings as $index => $tax_setting ) {
+				if ( in_array( $tax_setting['id'], array( 'tax_options', 'woocommerce_tax_classes', self::OPTION_NAME ) ) ) {
+					continue;
+				}
+				$tax_settings[$index]['custom_attributes'] = array( 'disabled' => true );
+			}
+		}
+
 		return $tax_settings;
 	}
 
+	/**
+	 * When automated taxes are enabled, overwrite core tax settings that might break the API integration
+	 * This is similar to the original plugin functionality where these options were reverted on page load
+	 * See: https://github.com/taxjar/taxjar-woocommerce-plugin/blob/82bf7c58/includes/class-wc-taxjar-integration.php#L66-L91
+	 *
+	 * @param mixed $value - option value
+	 * @param array $option - option metadata
+	 * @return string new option value, based on the automated taxes state or $value
+	 */
+	public function sanitize_tax_option( $value, $option ) {
+		if ( ! is_array( $option ) || ! $this->is_enabled() && 'yes' != $_POST[self::OPTION_NAME] ) {
+			return $value;
+		}
+
+		switch( $option['id'] ) {
+			case 'woocommerce_calc_taxes':
+				// If automated taxes are enabled and user disables taxes we re-enable them
+				return 'yes';
+			case 'woocommerce_tax_based_on':
+				// Users can set either billing or shipping address for tax rates but not shop
+				return $value === 'billing' ? $value : 'shipping';
+			case 'woocommerce_prices_include_tax':
+				// Rate calculations assume tax not included
+				return 'no';
+			case 'woocommerce_shipping_tax_class':
+				// Use no special handling on shipping taxes, our API handles that
+				return '';
+			case 'woocommerce_tax_round_at_subtotal':
+				// API handles rounding precision
+				return 'no';
+			case 'woocommerce_tax_display_shop':
+				// Rates are calculated in the cart assuming tax not included
+				return 'excl';
+			case 'woocommerce_tax_display_cart':
+				// TaxJar returns one total amount, not line item amounts
+				return 'excl';
+			case 'woocommerce_tax_total_display':
+				// TaxJar returns one total amount, not line item amounts
+				return 'single';
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Hack to hide the tax sections for additional tax class rate tables.
+	 */
+	public function output_sections_before() {
+		if ( ! $this->is_enabled() ) {
```

Comment:
Rather than check enabled status here, can we move the hook attachment to after the check in `init()`? _(Same for the _after() callback too)_